torch/jit/__init__.py

```diff
@@ -399,6 +402,33 @@ def script_method(fn):
     return ScriptMethodStub(createResolutionCallback(frames_up=2), get_jit_ast(fn), fn)
 
 
+class batch(object):
```

Comment:
Can you capitalize this class name?

Reply:
Then the decorator will be `@torch.jit.Batch`?

Reply:
Oh I see what's happening now. Doing @torch.jit.batch is creating an instance of this class where the __call__ method is applied to the passed in function. You can also just make this a function and not store away `batch_size`, `optimize` and `_frames_up` in class variables, just close over them in the nested `def`. I think that's what generally happens with decorators in PyTorch and would make the naming consistent with the style